test/e2e/helpers.go

```
@@ -0,0 +1,918 @@
+// restoreNetworkPolicyWithTimeout deletes a NetworkPolicy and waits for it to be restored with a custom timeout.
+func restoreNetworkPolicyWithTimeout(ctx context.Context, client kubernetes.Interface, expected *networkingv1.NetworkPolicy, timeout time.Duration) {
+	g.GinkgoHelper()
+	namespace := expected.Namespace
+	name := expected.Name
+	g.GinkgoWriter.Printf("deleting NetworkPolicy %s/%s\n", namespace, name)
+	err := client.NetworkingV1().NetworkPolicies(namespace).Delete(ctx, name, metav1.DeleteOptions{})
+	o.Expect(err).NotTo(o.HaveOccurred(), "failed to delete NetworkPolicy %s/%s", namespace, name)
+	err = wait.PollUntilContextTimeout(ctx, 5*time.Second, timeout, true, func(ctx context.Context) (bool, error) {
+		current, err := client.NetworkingV1().NetworkPolicies(namespace).Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			return false, nil
```

golangci-lint (2.11.4)

[error] error is not nil (line 316) but it returns nil

(nilerr)